### ISSUE TITLE
Attributes aren't written to file as expected

### ISSUE THREAD

**Issue by rstoneback** [pysat/pysat#276]
**Describe the bug**
I set some custom attributes to an Instrument object. When I write that object to a netCDF file the attributes are out of date. Values written to files are those set previously by load_netcdf.

**To Reproduce**
1. Load a file, created by pysat, using load_netcdf that has custom attributes
2. Set attributes with the same name directly to Instrument object
3. Write a file using to_netcdf

**Expected behavior**
I expected the custom attributes I set upon the object to end up in the file.

**Screenshots**

**Desktop (please complete the following information):**
 - OS:MacOS
 - 10.14.5

**Additional context**
What is likely happening is the custom attributes are assigned to the metadata object when loaded. I then set new attributes to Instrument. The order of ops in to_netcdf writes Instrument attributes then MetaData attributes. Older attributes in metadata are rewriting values in the netcdf file. We could simply switch the order, but that leads to new bugs.

I expected those custom meta attributes to be transferred to the Instrument object within pysat, automatically. Somewhere in the general load routine. Check into this.

Should also produce a warning, or perhaps even an error, if there are attributes at both Instrument and MetaData level when writing a netCDF4 file.

**Comment by rstoneback** [comment]
There should be a method in .meta that copies attributes from meta over to the instrument object. This should probably be placed in the user facing load routine.

**Comment by asher-pembroke** [comment]
I'm adding the following test to `test_utils.py/TestBasicNetCDF4`

```python
def test_netcdf_meta_change(self):

    self.testInst.load(2009, 1)
    current_units = self.testInst.meta['uts', 'units']

    new_units = current_units + '^2'

    self.testInst.meta['uts', 'units'] = new_units
    
    outfile = os.path.join(self.testInst.files.data_path,
                           'pysat_test_ncdf.nc')

    self.testInst.to_netcdf4(outfile)

    loaded_inst, meta = pysat.utils.load_netcdf4(outfile)

    try:
        assert meta['uts','units'] == new_units
    except AssertionError:
        print('{} != {}'.format(meta['uts','units'], new_units))
        raise
```
I think this test captures the expected behavior.

**Comment by asher-pembroke** [comment]
I'm still a little confused about the metadata vs attribute model and how that differs in the netcdf case. Can someone provide a test that illustrates the problem for me?

**Comment by asher-pembroke** [comment]
I'm getting a different problem from what you describe

```python
testInst = pysat.Instrument(platform='pysat',
                                 name='testing',
                                 clean_level='clean')
testInst.pandas_format = True

testInst.load(2009, 1)

try:
    assert testInst.bespoke # should raise
except AttributeError as m:
    print(m)
```
```console
Returning pysat testing data for 01 January 2009
'Instrument' object has no attribute 'bespoke'
```
```python

fname = testInst.files.files[0]
testInst.meta.bespoke = True

testInst.meta.transfer_attributes_to_instrument(testInst)

# custom meta attribute correctly assigned to instrument
assert testInst.bespoke

outfile = os.path.join(testInst.files.data_path, fname)
testInst.to_netcdf4(outfile)

loaded_inst, meta = pysat.utils.load_netcdf4(outfile)

# custom attribute correctly read from file
assert meta.bespoke
```
So it looks like this works as expected, but if I try to load the instrument then the custom attribute isn't set at all.

```python
print('creating new instrument')
testInst = pysat.Instrument(platform='pysat',
                             name='testing',
                             clean_level='clean')
testInst.load(fname = fname)

# custom attribute not assigned to instrument
try:
    assert testInst.meta.bespoke == True
except AttributeError as m:
    print(m)
```
```console
creating new instrument
Returning pysat testing data from 2008-01-01.nofile
'Meta' object has no attribute 'bespoke'
```
Am I testing this error properly?

**Comment by jklenzing** [comment]
Closed with merge of #318.
